server/channels/api4/user.go

```diff
@@ -3445,3 +3446,59 @@ func getUsersWithInvalidEmails(c *Context, w http.ResponseWriter, r *http.Reques
 		c.Logger.Warn("Error writing response", mlog.Err(err))
 	}
 }
+
+func resetUserAuthDataToEmail(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSystem) {
+		c.SetPermissionError(model.PermissionManageSystem)
+		return
+	}
+	type ResetAuthDataParams struct {
+		AuthService      string   `json:"auth_service"`
+		IncludeDeleted   bool     `json:"include_deleted"`
+		DryRun           bool     `json:"dry_run"`
+		SpecifiedUserIDs []string `json:"user_ids"`
+	}
+	var params *ResetAuthDataParams
+	jsonErr := json.NewDecoder(r.Body).Decode(&params)
+	if jsonErr != nil || params == nil {
+		c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "model.utils.decode_json.app_error", nil, "", http.StatusBadRequest).Wrap(jsonErr)
+		return
+	}
+
+	authService := params.AuthService
+
+	// Check the auth service matches our offerings
+	if authService == "" || (authService != model.UserAuthServiceSaml && authService != model.ServiceOpenid && authService != model.UserAuthServiceGitlab && authService != model.ServiceGoogle && authService != model.ServiceOffice365) {
+		c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "api.admin.users.invalid_auth_type.app_error", map[string]any{"AuthService": authService}, "", http.StatusBadRequest)
```

Comment:
Nit: A specific error message when authService is empty would be helpful for anyone using the API.